### DISCUSSION TITLE
Process multiple OpenAPI yaml files in Gradle

### DISCUSSION THREAD

**Discussion by davidjohnstonverint** [openapi-processor/openapi-processor-spring#137] [Q&A] [answered]
I have a project in which we have two different OpenAPI specs (one for the main API and another is a private tenant admin API that we want to keep separate from the main API for reasons). I want to have a single Java project that contains the codebase for both of these specs, is there a way in which we can configure the gradle plugin to process two separate yaml files? I would like to essentially have two different input yaml files that generate code to differently configured output directories.  Any suggestions would be appreciated.

**Comment by hauner** [comment]
Quick answer, I have to check if this really works. :-)

In theory something like this could work:

```groovy
openapiProcessor {
    spring {
        apiPath "${projectDir}/src/api/openapi.yaml"
        processor 'io.openapiprocessor:openapi-processor-spring:<version>'
        targetDir "${projectDir}/build/openapi"
        mapping "${projectDir}/src/api/mapping.yaml"
    }
    spring2 {
        apiPath "${projectDir}/src/api/openapi2.yaml"
        processor 'io.openapiprocessor:openapi-processor-spring:<version>'
        targetDir "${projectDir}/build/openapi2"
        mapping "${projectDir}/src/api/mapping2.yaml"
    }
}
```

you would also have to add a second `dependsOn` for spring2. 

```compileJava.dependsOn ('processSpring2')```

may be best to have distinct folders for the `targetDir`, not sure what happens if both (spring/spring2) work on the same target.

**Reply by davidjohnstonverint** [reply]
Unfortunately that causes an error:

Cause: can't find processor: spring2!

Is that an error from openapiprocessor or gradle? Is the configuration block called spring because that is specifying that it is using the spring processor (even though the processor is specified inside the block), and so another block called spring2 makes it think it is to use a different processor other than spring?

**Reply by hauner** [reply]
ok,  I have to check this. I thought it should work. :-(

Would this generally be a practical way to support multiple apis? Or would you prefer another way, like allowing a list for `apiPath`?

**Reply by davidjohnstonverint** [reply]
It doesn't have to be this way, another way like you said of specifying multiple apiPath/targetDir properties would also work.

**Comment by hauner** [comment, marked as answer]
My original suggestion wasn't too far off. It was missing the `name` override. Unfortunately there was a small bug that broke it at another place. :-(

It is fixed in the new plugin version [2022.1](https://github.com/openapi-processor/openapi-processor-gradle/releases/tag/v2022.1) . However, it requires gradle 7+.  I hope that is not an issue.  

```groovy
openapiProcessor {
    spring2 {
        processorName 'spring'
        /// ....
    }
}
```

**Reply by davidjohnstonverint** [reply]
Perfect, thank you.
